pkg/system/phase2_creating.go

```diff
@@ -75,6 +86,21 @@ func (r *Reconciler) SetDesiredServiceAccount() {
 		`{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"` + r.RouteMgmt.Name + `"}}`
 }
 
+// ReconcileNoobaaDB reconciles NoobaaDB statefulset
+func (r *Reconciler) ReconcileNoobaaDB() error {
+	if err := r.ReconcileObject(r.NoobaaDB, r.SetDesiredNoobaaDB); err != nil {
+		return err
+	}
+	// check if mongoDB is ready. if not return error
+	if !util.KubeCheck(r.NoobaaDBPod) {
```

Comment:
Why do we need to wait for the pod before we can continue?
There are plenty of things to create in phase2, and better create it all and then wait for all to start serving in phase3.